src/System.Private.CoreLib/shared/System/MemoryExtensions.Trim.cs

```diff
@@ -0,0 +1,818 @@
+            Debug.Assert((uint)start <= span.Length);
+
+            int end = span.Length - 1;
+
+            if (trimElement == null)
+            {
+                for (; end >= start; end--)
+                {
+                    if (span[end] != null)
+                        break;
+                }
+            }
+            else
+            {
+                for (; end >= start; end--)
+                {
+                    if (!trimElement.Equals(span[end]))
+                        break;
+                }
+            }
+
+            return end - start + 1;
+        }
+
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a set of elements specified
+        /// in a readonly span from the memory.
+        /// </summary>
+        /// <param name="trimElements">The span which contains the set of elements to remove.</param>
+        /// <remarks>If <paramref name="trimElements"/> is empty, the memory is returned unaltered.</remarks>
+        public static Memory<T> Trim<T>(this Memory<T> memory, ReadOnlySpan<T> trimElements)
+            where T : IEquatable<T>
+        {
+            if (trimElements.Length <= 1) // Optimize for N > 1
+            {
+                return trimElements.Length == 0 ? memory : Trim(memory, trimElements[0]);
+            }
+
+            Span<T> span = memory.Span;
+            int start = ClampStart(span, trimElements);
+            int length = ClampEnd(span, start, trimElements);
+            return memory.Slice(start, length);
+        }
+
+        /// <summary>
+        /// Removes all leading occurrences of a set of elements specified
+        /// in a readonly span from the memory.
+        /// </summary>
+        /// <param name="trimElements">The span which contains the set of elements to remove.</param>
+        /// <remarks>If <paramref name="trimElements"/> is empty, the memory is returned unaltered.</remarks>
+        public static Memory<T> TrimStart<T>(this Memory<T> memory, ReadOnlySpan<T> trimElements)
+            where T : IEquatable<T>
+        {
+            if (trimElements.Length <= 1) // Optimize for N > 1
+            {
+                return trimElements.Length == 0 ? memory : TrimStart(memory, trimElements[0]);
+            }
+
+            int start = ClampStart(memory.Span, trimElements);
+            return memory.Slice(start);
+        }
+
+        /// <summary>
+        /// Removes all trailing occurrences of a set of elements specified
+        /// in a readonly span from the memory.
+        /// </summary>
+        /// <param name="trimElements">The span which contains the set of elements to remove.</param>
+        /// <remarks>If <paramref name="trimElements"/> is empty, the memory is returned unaltered.</remarks>
+        public static Memory<T> TrimEnd<T>(this Memory<T> memory, ReadOnlySpan<T> trimElements)
+            where T : IEquatable<T>
+        {
+            if (trimElements.Length <= 1) // Optimize for N > 1
```

Comment:
Should this say Optimize for `N <= 1`?

Reply:
Hm, I guess that comment is confusing. What I meant is, instead of checking via two `if`s, I nest them since they're both unlikely:
```csharp
if (len <= 1)
    return len == 0 ? x : y;
```
vs
```csharp
if (len == 0)
    return x;
else if (len == 1)
    return y;
```

Reply:
Right, that makes sense. If we are resetting CI anyway, I would clarify the comment with essentially what you said here. Something like: `Add nested checks to avoid unnecessary branching for the uncommon case of N <= 1`

Reply:
Fixed